### ISSUE TITLE
"Failed to create data source 'linux.ftrace'" 

### ISSUE THREAD

**Issue by AartiPhatke** [google/perfetto#511]
Hello, I was trying to run a trace using a provided config file in the config folder:  
./perfetto --txt -c /mnt/configs/ftrace.cfg -o trace_file.perfetto-trace

I did the set up for perfetto slightly differently since I wanted to run it on a ARMv7 system. I cross compiled it using gcc-linaro-7.2.1-2017.11-x86_64_arm-linux-gnueabihf and followed the Perfetto docs section "Cross-compiling for Linux ARM/64".

However I am getting the following errors: 
![Capture](https://github.com/google/perfetto/assets/98285237/8134b219-c292-4443-8abd-dd7250759bb7)

root@SWAA31J001:/mnt# ./perfetto --txt -c /mnt/configs/ftrace.cfg -o trace_file2.
.perfetto-trace
[380.817]     perfetto_cmd.cc:609 Parsing TraceConfig, 11280 bytes
[381.263]    perfetto_cmd.cc:1051 Connected to the Perfetto traced service, TTL: 10s
[381.292] ing_service_impl.cc:450 Consumer 0x580270 connected from UID 0
[381.297] ing_service_impl.cc:550 Enabling tracing for consumer 0x5802f0, UUID: d7702b89-f3e0-8f82-464c-c0908cdd1c7c
[381.303] ng_service_impl.cc:2789 Setting up data source linux.ftrace with target buffer 2
[381.304] ng_service_impl.cc:2789 Setting up data source linux.process_stats with target buffer 2
[381.306] ng_service_impl.cc:2789 Setting up data source linux.inode_file_map with target buffer 2
[381.307] ng_service_impl.cc:2789 Setting up data source perfetto.metatrace with target buffer 2
[381.309] ng_service_impl.cc:1003 Configured tracing session 2, #sources:4, duration:10000 ms, #buffers:1, total buffer size:65536 KB, total sessions:1, uid:0 session name: ""
[381.343]  probes_producer.cc:395 SetupDataSource(id=5, name=linux.ftrace)
**[381.343]  probes_producer.cc:411 Failed to create data source 'linux.ftrace'**
[381.355]  probes_producer.cc:395 SetupDataSource(id=6, name=linux.process_stats)
[381.385]  probes_producer.cc:395 SetupDataSource(id=7, name=linux.inode_file_map)
[381.392]  probes_producer.cc:159 Inode file map setup (target_buf=2)
[381.393]      file_scanner.cc:84 opendir /system (errno: 2, No such file or directory)
[381.431]  probes_producer.cc:395 SetupDataSource(id=8, name=perfetto.metatrace)
[381.442]  probes_producer.cc:423 StartDataSource(id=5, name=linux.ftrace)
**[381.454]  probes_producer.cc:427 Data source id=5 not found**
[381.480]  probes_producer.cc:423 StartDataSource(id=6, name=linux.process_stats)
[381.483]  probes_producer.cc:423 StartDataSource(id=7, name=linux.inode_file_map)
[381.489]  probes_producer.cc:423 StartDataSource(id=8, name=perfetto.metatrace)
^C[391.115]    perfetto_cmd.cc:1265 SIGINT/SIGTERM received: disabling tracing.
[391.129]  probes_producer.cc:544 Flush 2 acked by data source 6
[391.129]  probes_producer.cc:544 Flush 2 acked by data source 7
[391.130]  probes_producer.cc:544 Flush 2 acked by data source 8
[391.130]  probes_producer.cc:556 All data sources acked to flush 2
[391.142]  probes_producer.cc:454 Producer stop (id=5)
**[391.142]  probes_producer.cc:458 Cannot stop data source id=5, not found**
[391.145]  probes_producer.cc:454 Producer stop (id=6)
[391.151]  probes_producer.cc:454 Producer stop (id=7)
[391.157]  probes_producer.cc:454 Producer stop (id=8)
[391.314] ng_service_impl.cc:1965 Triggering final flush for 2
**[391.314] ng_service_impl.cc:1733 Flush() called, but tracing has not been started**
[391.314] ng_service_impl.cc:1970 FlushAndDisableTracing(2) done, success=0
^C
root@SWAA31J001:/mnt# [393.528] ing_service_impl.cc:467 Consumer 0x5802f0 disconnected
[393.529] ng_service_impl.cc:2478 Freeing buffers for session 2
[393.530] ng_service_impl.cc:2510 Tracing session 2 ended, total sessions:0

**Comment by primiano** [comment]
That's a failure while initializing the ftrace module.

Very likely this is either:

1) A permission issue, check that the current user has permission to access /sys/kernel/tracing (but you seem root, so I'd rule this out unless you have selinux or some other LSM module enabled)

2) A failure due to the fact that something else is using ftrace / or tracing_on is 1 (reset it first)

3) some other ftrace parse error (rebuild in debug mode or with `perfetto_force_dlog = true` to find out

**Comment by AartiPhatke** [comment]
How would you do 2. ? I tried to set tracing_on to 0 `echo 0 > /sys/kernel/debug/tracing/tracing_on` but that the tracing folder does not exist. 
Could you also walk me though the steps for 3. ?

**Comment by LalitMaganti** [comment]
Are you sure your build is compiled with tracefs enabled? On new kernels
the path should be /sys/kernel/tracing, on old /sys/kernel/debug/tracing.

**Comment by AartiPhatke** [comment]
This is the kernel version I'm using: `Linux SWAA31J001 4.9.126-g259f2e0422 #1 PREEMPT Wed May 18 21:18:56 UTC 2022 armv7l GNU/Linux`. I would assume these paths would exist but they don't, how do I add them in?

**Comment by LalitMaganti** [comment]
Well you kernel might not even be complied with debugfs/tracefs enabled.
I'd start by checking that first. Next I'd see if you can mount tracefs at
one of those paths.

But we're also sort of getting out of the realm of the sort of things we
can really help with. Custom compilation of kernels really has too many
variables and is not really in the area of our expertise.

**Comment by primiano** [comment]
Yeah feels like tracefs is either not built in the kernel (see CONFIG_TRACING) or is not mounted.
Try `mount -t tracefs nodev /sys/kernel/tracing`

Closing this issue as everything seems working as intended in perfetto
